instrumentation/kamon-cats-io/src/test/scala/kamon/instrumentation/cats/io/AbstractCatsEffectInstrumentationSpec.scala

```diff
@@ -0,0 +1,115 @@
+package kamon.instrumentation.cats.io
+
+import java.util.concurrent.Executors
+
+import cats.effect.{Async, ContextShift, Effect, LiftIO, Timer}
+import cats.implicits._
+import kamon.Kamon
+import kamon.context.Context
+import kamon.instrumentation.cats.io.Tracing.Implicits._
+import kamon.tag.Lookups.plain
+import kamon.testkit.TestSpanReporter
+import kamon.trace.Span
+import org.scalatest.concurrent.{Eventually, PatienceConfiguration, ScalaFutures}
+import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, Inspectors, Matchers, WordSpec}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContext.global
+import scala.concurrent.duration.Duration
+
+// NOTE: We have this test just to ensure that the Context propagation is working, but starting with Kamon 2.0 there
+//       is no need to have explicit Runnable/Callable instrumentation because the instrumentation brought by the
+//       kamon-executors module should take care of all non-JDK Runnable/Callable implementations.
+abstract class AbstractCatsEffectInstrumentationSpec[F[_]: LiftIO](effectName: String)(implicit F: Effect[F])
+  extends WordSpec
+    with ScalaFutures
+    with Matchers
+    with PatienceConfiguration
+    with TestSpanReporter
+    with Inspectors
+    with Eventually
+    with BeforeAndAfterAll
+    with BeforeAndAfter {
+
+  implicit def contextShift: ContextShift[F]
+
+  implicit def timer: Timer[F]
+
+  private val customExecutionContext = ExecutionContext.fromExecutorService(Executors.newCachedThreadPool())
```

Comment:
I'd go with a fixed ThreadPool to ensure the test are running with multiple threads